panimg/image_builders/dicom.py

```diff
@@ -59,7 +45,281 @@ def _find_dicom_tag(dataset: pydicom.Dataset, tag: str):
     )
 
 
-def _get_headers_by_study(files, file_errors):
+class DicomDataset:
+    def __init__(self, index, headers, n_time, n_slices, n_slices_per_file):
+        self.index = index
+        self.headers = headers
+        self.n_time = n_time
+        self.n_slices = n_slices
+        self.n_slices_per_file = n_slices_per_file
+
+        self._pixel_value_dtype = None
+
+    @property
+    def ref_header(self) -> pydicom.Dataset:
+        return self.headers[0]["data"]
+
+    @property
+    def dimensions(self) -> int:
+        # Images are either 4D or 3D (2D images get an additional axis of size 1)
+        return 4 if self.n_time and self.n_time > 1 else 3
+
+    def direction(self) -> np.ndarray:
```

Comment:
Is there a design choice behind not making this a property?

Reply:
Hm, I did not make it a property because it does not just return a value but needs to first calculate the rotation matrix. But makes sense to turn this also into a property and store the matrix in `self._direction` so that it is only calculated once. Are there any good rules of thumb when you would use a property? Have to admit that I often use them a bit arbitrarily.

Reply:
Neat! Unfortunately the package still supports Python 3.7 and `cached_property` was added only in 3.8. I've changed it anyway using a variable.

Reply:
Ah yes, just noticed =/.